python/ray/data/tests/test_dataset_formats.py

```diff
@@ -1290,6 +1292,17 @@ def test_numpy_read(ray_start_regular_shared, tmp_path):
     assert [v.item() for v in ds.take(2)] == [0, 1]
 
 
+def test_numpy_read_partitioning(ray_start_regular_shared, tmp_path):
+    path = os.path.join(tmp_path, "country=us", "data.npy")
+    os.mkdir(os.path.dirname(path))
+    np.save(path, np.arange(4).reshape([2, 2]))
+
+    ds = ray.data.read_numpy(path, partitioning=Partitioning("hive"))
+
+    assert ds.schema().names == ["__value__", "country"]
```

Comment:
Agreed about the schema, ideally we would return some kind of `TensorSchema` that houses the element shape and dtype, where the column name is hidden. For now we've been living with that leak since it's only when inspecting metadata, not when e.g. creating UDFs; the latter would be a much larger UX impact.

I think that we should adopt the human-readable column name when partitioning is present, since it's now in a tabular form. When partitioning is not present, that `"__value__"` column name will eventually be completely hidden from the user (currently I think it's only leaked in the schema/repr), so there shouldn't be any confusion around changing the column name, since even with that current leak, they never interacted with the `"__value__"` column in code anyway.

Reply:
> "__value__" column name will eventually be completely hidden from the user (currently I think it's only leaked in the schema/repr)

Ah, wasn't aware of this.

Sgtm.